src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -455,6 +430,44 @@ private Collection<FunctionDescriptor> ReadFunctions(ScriptHostConfiguration con
             return ReadFunctions(metadatas, descriptorProviders);
         }
 
+        /// <summary>
+        /// Determines which script should be considered the "primary" entry point script.
+        /// </summary>
+        internal static string DeterminePrimaryScriptFile(JObject functionConfig, string[] functionFiles)
+        {
+            if (functionFiles.Length == 1)
+            {
+                // if there is only a single file, that file is primary
+                return functionFiles[0];
```

Comment:
I know it would be a breaking change, but looking back, I wish we didn't have this special case at all, as it results in a cliff. I'd rather it fail right away when users only have one file, then mysteriously start failing when you add a second one.